comps/llms/README.md

```diff
@@ -6,111 +6,123 @@ A prerequisite for using this microservice is that users must have a LLM text ge
 
 Overall, this microservice offers a streamlined way to integrate large language model inference into applications, requiring minimal setup from the user beyond initiating a TGI/vLLM/Ray service and configuring the necessary environment variables. This allows for the seamless processing of queries and documents to generate intelligent, context-aware responses.
 
+## Clone OPEA GenAIComps
+
+Clone this repository at your desired location and set an environment variable for easy setup and usage throughout the instructions.
+
+```bash
+git clone https://github.com/opea-project/GenAIComps.git
+
+export OPEA_GENAICOMPS_ROOT=/path/to/GenAIComps
+```
+
 ## 🚀1. Start Microservice with Python (Option 1)
 
 To start the LLM microservice, you need to install python packages first.
 
 ### 1.1 Install Requirements
 
 ```bash
-pip install -r requirements.txt
+pip install opea-comps
+pip install -r ${OPEA_GENAICOMPS_ROOT}/comps/llms/requirements.txt
+
+# Install requirements of your choice of microservice in the text-generation folder (tgi, vllm, vllm-ray, etc.)
+export MICROSERVICE_DIR=your_chosen_microservice
+
+pip install -r ${OPEA_GENAICOMPS_ROOT}/comps/llms/text-generation/${MICROSERVICE_DIR}/requirements.txt
 ```
 
-### 1.2 Start LLM Service
+### 1.2 Start LLM Service with Python Script
 
-#### 1.2.1 Start TGI Service
+#### 1.2.1 Start the TGI Service
 
 ```bash
-export HF_TOKEN=${your_hf_api_token}
-export LANGCHAIN_TRACING_V2=true
-export LANGCHAIN_API_KEY=${your_langchain_api_key}
-export LANGCHAIN_PROJECT="opea/gen-ai-comps:llms"
-docker run -p 8008:80 -v ./data:/data --name tgi_service --shm-size 1g ghcr.io/huggingface/text-generation-inference:1.4 --model-id ${your_hf_llm_model}
+export TGI_LLM_ENDPOINT="http://${your_ip}:8008"
```

Comment:
As far as I know, its supposed to be the IP address of the machine from where you are consuming the microservice, so it could be the same machine you're running on or a different one.